### ISSUE TITLE
Usage of `allOf` generates an incomplete type

### ISSUE THREAD

**Issue by jogelin** [anymaniax/orval#191]
## What are the steps to reproduce this issue?
1. Clone your repo
2. `yarn install` and `yarn build` on the root
3. `cd samples/basic/ && yarn install` and modify the `petstore.yaml`
![image](https://user-images.githubusercontent.com/954509/130098405-918fbb3d-f24c-44a6-819c-2d8a3c8eb60d.png)
3. Generate the model using `yarn example`

## What happens?
The generated models look like this:

`petBase.ts`
```
/*
 * Generated by orval v5.5.3 🍺
 * Do not edit manually.
 * Swagger Petstore
 * OpenAPI spec version: 1.0.0
 */

export interface PetBase {
  id: number;
  name: string;
  tag?: string;
}

```

`pet.ts`
```
/*
 * Generated by orval v5.5.3 🍺
 * Do not edit manually.
 * Swagger Petstore
 * OpenAPI spec version: 1.0.0
 */
import type { PetBase } from './petBase';

export type Pet = PetBase;
```

## What were you expecting to happen?
I would expect `Pet` to contain all fields of `PetBase` and the new `color` property like this:

`pet.ts`
```
/*
 * Generated by orval v5.5.3 🍺
 * Do not edit manually.
 * Swagger Petstore
 * OpenAPI spec version: 1.0.0
 */
import type { PetBase } from './petBase';

export interface Pet extends PetBase {
  color: string;
}
// OR
export type Pet = PetBase & { color: string; };
```

**Comment by anymaniax** [comment]
Hello @jogeli to be honest I don't know if that a good way to write the allOf so I don't handle it like that. The way it's handle is like in the doc [here](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/).

**Comment by jogelin** [comment]
Hey @anymaniax, ok I understood my mistake, it was a syntax issue. I should specify it like:

```
    Pet:
      allOf:
        - $ref: '#/components/schemas/PetBase'
        - type: object <-- HERE
          required:
            - id
            - name
          properties:
            color:
              type: string
```

TBH, I would like to migrate from `openapi generator`. In openapi, the generation was working even with the wrong syntax:
```
export interface Pet {
  id: number;
  name: string;
  tag?: string;
  color: string;
}
```

I'll fix my specs, thank you ;)

**Comment by anymaniax** [comment]
If you have other problems don't hesitate to open another issue 😀. Improvement ideas are also welcome 🚀

**Comment by fnatte** [comment]
Hi @anymaniax, I encountered this issue as well, and I'm not sure whether to open another issue or not since it's pretty much the same case.

I'm using https://github.com/domaindrivendev/Swashbuckle.AspNetCore to generate the API spec, therefore it's not an easy fix to change the spec structure as the solution above.

Swashbuckle does inheritance like:
```
Circle: {
  type: "object",
  allOf: [
    {
      $ref: "#/components/schemas/Shape"
    }
  ],
  properties: {
    radius: {
      type: "integer",
      format: "int32",
    }
  },
},
Shape: {
  type: "object",
  properties: {
    name: {
      type: "string",
      nullable: true,
    }
  },
}
```
(https://github.com/domaindrivendev/Swashbuckle.AspNetCore#enabling-inheritance)

Looks to me like there are two competing syntaxes here. You are probably correct about your implementation of `allOf` inheritance (being backed by [the doc you linked ](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/)). However, I'm wondering if it's an option to also be compatible with the Swashbuckle-way?

I'll do some thinking of my options here. What're your thoughts on this?

**Comment by anymaniax** [comment]
Hello @fnatte, I will take a look and see what I can do. In the meantime, you can use a transformer to modify the specification if you want.

**Comment by fnatte** [comment]
Using the following transform for now:
```javascript
const transformAllOfInheritence = (inputSchema) => ({
  ...inputSchema,
  components: {
    ...inputSchema.components,
    schemas: inputSchema.components.schemas
      ? Object.entries(inputSchema.components.schemas).reduce((acc, [schemaName, schema]) => {
          if (schema.allOf && schema.properties) {
            const { allOf, description, ...rest } = schema;
            const modifiedSchema = {
              description,
              allOf: [...allOf, { ...rest }],
            };
            return { ...acc, ...{ [schemaName]: modifiedSchema } };
          }

          return { ...acc, ...{ [schemaName]: schema } };
        }, {})
      : undefined,
  },
});

module.exports = (inputSchema) => transformAllOfInheritence(inputSchema);
```

**Comment by anymaniax** [comment]
will be added with the next version

**Comment by anymaniax** [comment]
Should be fixed with `6.4.0` can you test when you have time, please?

**Comment by anymaniax** [comment]
Closing this issue since it's normally fixed. Don't hesitate if you have any problem
